discrete_optimization/coloring/solvers/coloring_toulbar_solver.py

```diff
@@ -0,0 +1,166 @@
+#  Copyright (c) 2022 AIRBUS and its affiliates.
+#  This source code is licensed under the MIT license found in the
+#  LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+from discrete_optimization.coloring.coloring_model import (
+    ColoringProblem,
+    ColoringSolution,
+)
+from discrete_optimization.coloring.solvers.greedy_coloring import (
+    GreedyColoring,
+    NXGreedyColoringMethod,
+)
+from discrete_optimization.generic_tools.do_problem import (
+    ParamsObjectiveFunction,
+    build_aggreg_function_and_params_objective,
+)
+from discrete_optimization.generic_tools.do_solver import SolverDO
+from discrete_optimization.generic_tools.result_storage.result_storage import (
+    ResultStorage,
+)
+
+try:
+    import pytoulbar2
+except ImportError:
+    toulbar_available = False
+else:
+    toulbar_available = True
+
+
+class ToulbarColoringSolver(SolverDO):
+    def __init__(
+        self,
+        problem: ColoringProblem,
+        params_objective_function: Optional[ParamsObjectiveFunction] = None,
+    ):
+        self.problem = problem
+        (
+            self.aggreg_sol,
+            self.aggreg_from_dict_values,
+            self.params_objective_function,
+        ) = build_aggreg_function_and_params_objective(
+            self.problem, params_objective_function=params_objective_function
+        )
+        self.model: pytoulbar2.CFN = None
```

Comment:
This is not correct typing. Integrate my commits on nht/toulbar_coloring to be mypy compliant.

Reply:
done (normally ?!)